templates/pages/setup/webhook/webhook_security.html.twig

```diff
@@ -0,0 +1,199 @@
+{#
+ # ---------------------------------------------------------------------
+ #
+ # GLPI - Gestionnaire Libre de Parc Informatique
+ #
+ # http://glpi-project.org
+ #
+ # @copyright 2015-2023 Teclib' and contributors.
+ # @copyright 2003-2014 by the INDEPNET Development Team.
+ # @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ #
+ # ---------------------------------------------------------------------
+ #
+ # LICENSE
+ #
+ # This file is part of GLPI.
+ #
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # ---------------------------------------------------------------------
+ #}
+
+{% extends "generic_show_form.html.twig" %}
+{% import 'components/form/fields_macros.html.twig' as fields %}
+{% set params  = params ?? [] %}
+{% set rand_field = rand|default(random()) %}
+
+{% block form_fields %}
+   {% block more_fields %}
+
+      {{ fields.passwordField(
+         'secret',
+         item.fields['secret'],
+         __('Secret'),
+         field_options|merge({
+            is_disclosable: true,
+            can_regenerate: true,
+            helper: __('The webhook secret can be shared with a target server to allow it to validate requests are actually coming from the GLPI server and that the content was not modified between GLPI and the target.')
+         })
+      ) }}
+
+      {% if secret_already_used is not empty %}
+         {% set warning_secret_already_used %}
+            <div class="alert alert-warning align-items-center flex-column" role="alert">
+               <i class='ti ti-alert-triangle-filled'></i>
+               <span class='ms-2'> {{ __('Secret is already use by another webhooks') }}</span>
+               <span>
+                  {% for webhook in secret_already_used %}
+                     <li  class='ms-4'>{{ webhook.link|raw }}</li>
```

Comment:
Here we could prevent usage of `|raw`, either by passing an array of `Webhook` in `secret_already_used` to be able to use `webhook.getLink()`, either by passing and array of ids to be able to use `{{ get_item_link('Webhook', webhook_id) }}`.

Reply:
This part seems odd in general. I'm not sure it is a good idea to tell anyone which webhooks have the same secret.
It doesn't make any difference from a security standpoint to just say "Secret is already use by another webhooks" without the list.

Reply:
Is it really a problem to have webhooks that share the same secret? If you need to send multiple events for a single object (e.g. Ticket::add and Ticket::update), I guess the target application admin will probably ask you if it is possible to use a unique secret, because it will consider your two webhooks as the same source.

Also, since this is an auto-generated field, it is intentional to have the same secret for multiple webhooks. Therefore, we could consider that the webhooks administrator is aware of what he is doing and that he considers that in this specific case there is no risk in sharing the same secret.

Reply:
I don't think so, but this alert message was defined in the specs for this feature.

Reply:
I don't remember precisely why we wrote about the duplication alert.
But I think we can remove it. As said, normal usage will generate different secret for each webhooks. If someone really want to have one shared secret, I guess it doesn't really matter.